ui/components/app/toast-master/toast-master-selectors.ts

```diff
@@ -0,0 +1,118 @@
+import { PayloadAction } from '@reduxjs/toolkit';
+import { ReactFragment } from 'react';
+import { PRIVACY_POLICY_DATE } from '../../../helpers/constants/privacy-policy';
+import {
+  SURVEY_DATE,
+  SURVEY_END_TIME,
+  SURVEY_START_TIME,
+} from '../../../helpers/constants/survey';
+import { SHOW_NFT_DETECTION_ENABLEMENT_TOAST } from '../../../store/actionConstants';
+import { submitRequestToBackground } from '../../../store/background-connection';
+
+// TODO: get this into one of the larger definitions of state type
+type State = {
+  appState?: {
+    showNftDetectionEnablementToast?: boolean;
+  };
+  metamask?: {
+    newPrivacyPolicyToastClickedOrClosed?: boolean;
+    newPrivacyPolicyToastShownDate?: number;
+    onboardingDate?: number;
+    showNftDetectionEnablementToast?: boolean;
+    surveyLinkLastClickedOrClosed?: number;
+  };
+};
```

Comment:
Maybe this ad-hoc type intersection with `MetaMaskReduxState` will better align with the shape of the inputs we expect to be passed into these selectors at their call sites. 

```suggestion
type State = Omit<MetaMaskReduxState, 'appState'> & {
  appState: {
    showNftDetectionEnablementToast?: boolean;
  };
  metamask: {
    newPrivacyPolicyToastClickedOrClosed?: boolean;
    newPrivacyPolicyToastShownDate?: number;
    onboardingDate?: number;
    showNftDetectionEnablementToast?: boolean;
    surveyLinkLastClickedOrClosed?: number;
  };
};
```

- Fair warning, this will cause a lot of errors in the test files, which should be resolved with `@ts-expect-error` or `as unknown as` (the former would be preferred as we're intentionally passing in incomplete input).
- Making the top-level properties optional is probably unnecessary, as `state.metamask` is used in a lot of selectors without optional chaining.
- The `Omit` usage is because `MetaMaskReduxState['appState']` is currently typed as `never`, and for any type `T`, `never & T` is `never`. This in contrast to the `metamask` property, which achieves intersection without needing to be omitted.